### PR TITLE
Fix OpenMP race condition in inversion_utils.f90

### LINKED ISSUE #463: Test hybrid parallelism (OpenMP+MPI)

### PR DESCRIPTION
**Pull request by matt-frey**
Closes #463.

### CODE REVIEW

File: src/3d/inversion/inversion_utils.f90 [hunk 1]
```diff
@@ -253,14 +253,10 @@ subroutine init_inversion
             ! Tridiagonal arrays for the vertical vorticity component:
             htdv(0, :, :) = one / a0
             etdv(0, :, :) = -two * ap * htdv(0, :, :)
-            !$omp parallel shared(a0, ap, etdv, htdv, nz) private(iz) default(none)
-            !$omp do
             do iz = 1, nz-1
                 htdv(iz, :, :) = one / (a0(:, :) + ap * etdv(iz-1, :, :))
```
[the remark this answers is not in the document]
Comment (matt-frey) [reply src/3d/inversion/inversion_utils.f90:257]: Yes, we can do shared memory parallelism over x and y.

[the remark this answers is not in the document]
Comment (matt-frey) [reply src/3d/inversion/inversion_utils.f90:257]: I think we can postpone this for the moment as we anyway do not use the full potential of OpenMP. We should go through the code carefully in another PR.

[the remark this answers is not in the document]
Comment (sjboeing) [reply src/3d/inversion/inversion_utils.f90:257]: Agreed. I have also found OpenMP does not always help for "small problems" like here.

[the remark this answers is not in the document]
Comment (matt-frey) [reply src/3d/inversion/inversion_utils.f90:257]: I opened an issue for this #517.

**Review by sjboeing — APPROVED** [review]
Thanks for fixing this!

**Pull request merged**